New Ruin: ICW Stolen Base (Not horrifically destroyed on github edition)

Ratvarr opened the pull request:
## About The Pull Request
Adds a new ruin to Waste Planets, the ICW Era Cybersun Venture, that has had quite a few groups come by...

And this is not an extra PR merged too! Isn't that swell. I am in agony.

<details><summary>Screenshots</summary>
<p>

![image](https://github.com/user-attachments/assets/284d1527-1663-4488-ad8f-44420f2ed42e)

![image](https://github.com/user-attachments/assets/1fec18b4-f174-49ef-bb05-9eef76ade413)
It's a little big, so I have to split it between ingame screenshots
![image](https://github.com/user-attachments/assets/789b4410-fa80-48c7-b064-f8d9d4a3da02)

![image](https://github.com/user-attachments/assets/5c5ee2b5-f871-45f9-b6fe-555e69f3e0f6)

![image](https://github.com/user-attachments/assets/37c695c9-e2f4-4605-98fc-b0b7b8cfe00f)

![image](https://github.com/user-attachments/assets/3cc4cb71-d986-47e2-9a40-34ef142d684a)

</p>
</details> 

## Why It's Good For The Game

More ruins = Good
Intended to be difficult, but has some pretty good loot for the price. And some interesting fluff behind the story...

## Changelog

:cl:
add: Long range sensors have detected an unknown signal broadcasting over local GPS on the local long-term waste-world...
/:cl:

Ratvarr committed:
quick, in out 3 week ruin. 4 weeks later, 1 map corruption, my branch FUCKING IMPLODING and having to look in horror as 2 branches fucking combine after having 2 seconds of skill in VSC+Github is not how I wanted to spend this time. If this breaks I might die.

Ratvarr committed:
fixes mapdiff hopefully

Ratvarr committed:
and again

Ratvarr committed:
and again again

Ratvarr committed:
and again again again

Ratvarr committed:
and again again again again

Ratvarr committed:
and again again again again again

Ratvarr committed:
and again again again again again

Ratvarr committed:
Merge branch 'master' into hopefully-not-cursed-ICW-base-edition-2

Ratvarr committed:
Merge branch 'master' into hopefully-not-cursed-ICW-base-edition-2

Ratvarr committed:
housekeeping

Ratvarr committed:
Merge branch 'hopefully-not-cursed-ICW-base-edition-2' of https://github.com/Ratvarr/Shiptest-Torment-Nexus into hopefully-not-cursed-ICW-base-edition-2

Ratvarr committed:
i forget

Ratvarr committed:
AND AGAIN

Ratvarr committed:
why is this the thing i fixed. what is going on here

Ratvarr committed:
i refuse to say "final"

Ratvarr committed:
Merge branch 'master' into hopefully-not-cursed-ICW-base-edition-2

Ratvarr committed:
times i have had to fix this: 3

Gristlebee reviewed (changes requested):
A good first ruin, but there are some issues I'm seeing.

- Overall, the main issue I'm seeing in this room cohesion and structure placement. It's not very clear what some spaces are supposed to be, or you merged together rooms that would be better separate. An important thing to keep in mind is how would people "realistically" build these sites and spaces.

- The waiting room/med storage/ER? space would be better as individual bespoke rooms for each niche. It's quite large for a single room as well. You could move out the med supplies to the warehouse below the biohazard wing as one possible option.
![image](https://github.com/user-attachments/assets/70429707-8263-47f3-a9ab-48773ad15e7f)
- I'm not sure what this room was originally supposed to be. Trauma team ready room? Patient recovery? Dorms space for the trauma team? It's quite empty as well, which makes it hard to tell what it is.
![image](https://github.com/user-attachments/assets/6a038081-cd96-416d-893d-4e0db7aee66f)
- You're on the right track with the surgery room here. It's a concise space, and obvious what it's suppossed to be at a glance.
![image](https://github.com/user-attachments/assets/6f855f77-f634-4189-aa2b-b249278c6a81)
- Virology green decals is a nice touch for what looks like a biological research wing, but it doesn't really feel like a Cybersun space palate-wise. I'd suggest getting rid of the showerfloors in favour of normal white tiles. Play around with the decals a bit more and experiment. See about adding some Cybersun teal, and trying other patterns beside a checkered floor.
![image](https://github.com/user-attachments/assets/8b07c336-1121-4edf-8f68-7c536ffeb41c)

- I'd recommend cutting down the amount of machines in here. It's a dense block of machines with way to walk around them, which doesn't make sense from an ingame construction standpoint. Try rearranging them similar to a normal telecomms set up.
![image](https://github.com/user-attachments/assets/9dcaa1f8-c38c-4875-a8bc-becf86f563b7)
- I assume this was supposed to be a vault/self destruct room before being converted to an armory by the Frontiersmen. This is good, as you make use of common conventions from other maps to get your point across. 
![image](https://github.com/user-attachments/assets/e993cbe1-5653-44b9-8f76-330b0c57568b)
- I'm not sure about this building's placement at the edge of the compound however. From an in-universe standpoint, you would want secure areas closer to the center of the compound. In terms of ruin flow, it's also important to consider where you place major loot areas. Having them in more central areas mean players get to see and move through more of your ruin, and consequently experience more challenge.
- The armor pile on the rack in the vault isn't great. I would recommend separating these out into lockers/crates.
![image](https://github.com/user-attachments/assets/93c8e06a-367d-4a92-8cc6-8a7d745c5f18)
- You should also use reinforced plasteel walls for your vaults as they're secure areas.
![image](https://github.com/user-attachments/assets/9c1e86b6-a971-4e63-bd81-9f4cd6c2bb98)
- I'd recommend splitting the dorms here into two separate rooms for the sleeping space and cabinets and an EVA room for the SSUs and radiation gear.
![image](https://github.com/user-attachments/assets/b47cf122-fad9-4a89-9389-849d2e18358a)
- Get rid of the shower bear.
![image](https://github.com/user-attachments/assets/fb8aebf7-dee0-4ed3-a452-f3712fed4839)
- The tile colors between the eating area/kitchen blend together. I'd suggest either using decals to make a more obvious partition between them, or experiment with a different tile color for the kitchen.
![image](https://github.com/user-attachments/assets/96a7ec38-4be6-4c0b-962c-225b70ddf4a6)
- Doubled money spawner on the desk.
![image](https://github.com/user-attachments/assets/f34823d2-d21a-45fe-9a79-5304d8aa7ab1)
- Same here.
![image](https://github.com/user-attachments/assets/c30723f4-f6e9-43ba-b87c-7ca1ce11a138)
- Not sure about the carpet placement in the gambling den and how the edges are cut by the doors. Think about how carpets are usually cut as squares/rectangular. I'd suggest having the carpets be in smaller sections by the chairs/sofas and have the walkways be tiled.
![image](https://github.com/user-attachments/assets/3d0bb54d-8394-43c5-985e-4f1e684ffe7b)
- Floor safes you can't find without metaknowledge/looking at strongdmm are bad. Add some kind of hint.
![image](https://github.com/user-attachments/assets/301fc5e8-30c7-4645-9ddd-8c09dd8c897b)
- Put the radiation gear in a closet rather than a crate.
![image](https://github.com/user-attachments/assets/587ec186-67a6-43de-afcc-48057a61bfd8)
- I assume the building collapsed due to an explosive. Scatter some more scrap metal on top of the girders, break the tiles a bit more. Maybe put some old bombs to make it more clear. If it's collapsed as well, you could make the building open air potentially.
![image](https://github.com/user-attachments/assets/74d6d090-073b-4141-bb0d-d206034ef0c4)
- Move the inside cargo bay door buttons closer to the doors.
![image](https://github.com/user-attachments/assets/ff246d47-f2d0-4b36-8846-73c649bfd245)
- The contents of the warehouse are a bit odd as well? I'm not sure why a biohazard research site would have so many machine and computer frames. Think about the kind of things that would be useful to kind of place you styled the ruin as.
![image](https://github.com/user-attachments/assets/c87e660a-894d-41ab-924e-288399f4a413)
- Use a wasteplanet subtype tile for the walkways or make one for the tiles you're using. It increases loadtime and atmos cycles otherwise.
- You don't need to worry about having cyclelink helpers on the doors where you mapped AACs. They should link automatically.

Ratvarr committed:
yea

Ratvarr committed:
i'll show you the boss of this base.

hey buddy i think you got the wrong door, the ACLF club's two planets down. fuck you. ah fuck you anarchist man, maybe you and I should settle it right here in the ring if your think your that tough. oh im gonna kick your ass. alright man lets go why dont you get out of that rad stuff ill strip down under this and we'll settle it in the ring. get out of that uhh nukie outfit. yeah, smart ass. i'll show you the boss of this base.

Ratvarr committed:
uhhg i i gforoget sorggyy

Ratvarr committed:
cybersun 2

Ratvarr committed:
last bit'a mappin' before i throw 2gether da tiles

Ratvarr committed:
time for Testing

Ratvarr committed:
time for review

Ratvarr committed:
Update wasteplanet_icwbase.dmm

Ratvarr commented:
<details><summary>Updated screenshots</summary>
<p>

![image](https://github.com/user-attachments/assets/54fe0c74-cefc-4194-bdc5-0092cf207549)

![image](https://github.com/user-attachments/assets/a701d68d-bff8-49c2-b8e8-2f8e45b905ff)


</p>
</details> 

ready for re-review

Gristlebee reviewed (changes requested):
This is much better, though a few small things I'd like to point out.

You're going to need to make the other tiles in here wasteplanet subtypes if you're going to put wasteplanet tiles in here.
![image](https://github.com/user-attachments/assets/c6911b34-3aa4-4ecb-8bd0-0ab87ec087c8)
Non-planetary turf in the broken airlock here
![image](https://github.com/user-attachments/assets/f3a1e96c-5ed6-4a3d-936e-686d5a175291)
The basketball court is cute. We do have basketballs in code under this path if you want to add a ball so people can play in the acid rain.
![image](https://github.com/user-attachments/assets/dd5f6fa4-d278-4a18-94a0-9b1ade7b175f)
You can move out the radiation suits, internals and masks into the SSUs rather than leaving them empty. The crate should either be a radiation crate, or you can just make it a rack since it will be only just the medkit now.
![image](https://github.com/user-attachments/assets/43a64de7-f644-4de7-8374-22b458870763)
Medical crates don't automatically populate on spawn. You should put something inside it, or else it will spawn empty.
![image](https://github.com/user-attachments/assets/2cf2109b-f027-4445-acc2-673d738a2b41)
Map a stethoscope somewhere on the map, maybe in the surgical room for the safes in the ruin.

Ratvarr committed:
brace for shock brace for shock

Ratvarr committed:
Merge branch 'master' into hopefully-not-cursed-ICW-base-edition-2

Burning02 commented:
Not entirely sure if its intentional, but in the screenshots it looks like the money is that fake money that doesn't work (Because none of the Shiptest machines can take it). Probably would want to swap to one of the spacecash money spawners, instead of the fake one unless its intentional.

Burning02 commented:
> Not entirely sure if its intentional, but in the screenshots it looks like the money is that fake money that doesn't work (Because none of the Shiptest machines can take it). Probably would want to swap to one of the spacecash money spawners, instead of the fake one unless its intentional.

Nevermind, ignore this. For whatever reason, the sprites just... odd in the Strong screenshot

Gristlebee reviewed (changes requested):
Feedback after testing on live:
___

- Use non-diagonal walls on spots with lights. It looks odd otherwise.
- Looking at the code, and it turns inherit suit storage units won't auto-pickup radiation suits. I'd suggest either swapping these out for radiation suit SSUs, or change the code on the inherit SSUs to make them do so.
![image](https://github.com/user-attachments/assets/f3a42733-5f3b-49f0-90b1-8893c7f77ed7)
- Stray girder in the wall.
![image](https://github.com/user-attachments/assets/a2556b56-317d-4c3c-a00d-b840a959db37)
- The VI Private Troopers all spawn with commanders on their outfits. Cull that from their gear.
In a similar vein, they don't all need eguns here in the graves. I'd say cut it down to just the e-tar.
I also noted the graves don't protect from acid rain, and the corpses melted. Something to keep in mind if that's something you don't want.
![image](https://github.com/user-attachments/assets/955a1b06-36e2-4099-8afe-958a0a51b712)
- You could put some extra cash here as loot in the gambling den as loot.
![image](https://github.com/user-attachments/assets/23f057d6-b20d-49c9-b8d8-0e3dafca4646)

Ratvarr commented:
how the                                 fuck
![image](https://github.com/user-attachments/assets/7193f2a3-151e-4ec9-9ac0-5fd4ab80c868)
how    did a       girder get there       ??????

Gristlebee commented:
> how the fuck

I'm thinking it might potentially be a freak error and just the game being weird, I did notice another random girder embedded inside a computer in the ruin during the round it was tested. You might want to see about booting this up on a local and see if the girders still pop up. It's probably not your fault.

Also, remember to fix your linters when you can. It looks like you accidentally doubled up on lights and firedoors at these co-ordinates.
![image](https://github.com/user-attachments/assets/2ea87705-9859-4fe3-9d99-b98897893274)

Ratvarr committed:
i never expected maggot surgery

Ratvarr committed:
Merge branch 'hopefully-not-cursed-ICW-base-edition-2' of https://github.com/Ratvarr/Shiptest-Torment-Nexus into hopefully-not-cursed-ICW-base-edition-2

Ratvarr committed:
should fix linters

Ratvarr committed:
oug

Ratvarr committed:
Merge branch 'master' into hopefully-not-cursed-ICW-base-edition-2

Ratvarr committed:
okay i just had to put one final touch

Ratvarr committed:
please

Ratvarr committed:
very last last minute fixes

Ratvarr committed:
other things picked up by testing

Gristlebee reviewed (changes requested):
- Something you can do if you want is set the enviroment smash to 0 on the frontiersmen so they dont smash the sandbags they're behind.
![image](https://github.com/user-attachments/assets/d0206677-81d7-4ffa-8fdf-2d1117c77b41)
Set this guy to neuter so he doesn't drop his rifle. There's already an SKM to be looted in the vault.
![image](https://github.com/user-attachments/assets/908c7dc9-1414-4b1f-ac8a-b6c8aa7bcf35)

Ratvarr committed:
ok
